File: dota-zone/dotazone/src/main/java/br/com/dotazone/view/adapter/FeedNewsListAdapter.java
package br.com.dotazone.view.adapter;

import android.content.Context;
import android.content.Intent;
import android.graphics.Typeface;
import android.net.Uri;
import android.text.Html;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.View.OnClickListener;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import br.com.dotazone.DotaZoneBrain;
import br.com.dotazone.R;
import br.com.dotazone.model.listeners.Controllable;

public class FeedNewsListAdapter extends BaseAdapter implements Controllable {

    private Context mContext;
    private LayoutInflater mInflate;

    public FeedNewsListAdapter(Context context) {

        this.mContext = context;
        this.mInflate = (LayoutInflater) this.mContext.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
    }

    @Override
    public int getCount() {
        Log.d(DotaZoneBrain.INSTANCE.getTAG(), "getCount  " + DotaZoneBrain.INSTANCE.getRssItems().size() + "");
        return DotaZoneBrain.INSTANCE.getRssItems().size();
    }

    @Override
    public Object getItem(int position) {

        Log.d(DotaZoneBrain.INSTANCE.getTAG(), "getItem  " + position + "");
        return DotaZoneBrain.INSTANCE.getRssItems().get(position);
    }

    @Override
    public long getItemId(int positionId) {

        Log.d(DotaZoneBrain.INSTANCE.getTAG(), "getItemId  " + positionId + "");
        return positionId;
    }

    @SuppressWarnings("deprecation")
    @Override
    public View getView(final int position, View convertView, ViewGroup parent) {

        View view = convertView;
        Log.d("teste", "position in view  " + position + "");

        if (view == null) {

            view = mInflate.inflate(R.layout.feed_news_item, parent, false);

        }
        Typeface font = Typeface.createFromAsset(mContext.getAssets(), "Roboto-Light.ttf");

        TextView titleNews = (TextView) view.findViewById(R.id.textItemTitle);
        titleNews.setText(DotaZoneBrain.INSTANCE.getRssItems().get(position).getTitle());

        TextView hourNews = (TextView) view.findViewById(R.id.textItemHour);
        hourNews.setText(DotaZoneBrain.INSTANCE.getRssItems().get(position).getPubDate());

        TextView descriptionNews = (TextView) view.findViewById(R.id.textItemDescription);
        descriptionNews.setText(Html.fromHtml(DotaZoneBrain.INSTANCE.getRssItems().get(position).getDescription()));

        descriptionNews.setOnClickListener(new OnClickListener() {

            @Override
            public void onClick(View arg0) {

                Intent browserIntent = new Intent(Intent.ACTION_VIEW, Uri.parse(DotaZoneBrain.INSTANCE.getRssItems().get(position).getLink()));
                mContext.startActivity(browserIntent);

            }
        });

        titleNews.setTypeface(font);
        hourNews.setTypeface(font);
        descriptionNews.setTypeface(font);

        return view;
    }

    @Override
    public void menu(boolean isOpen) {
        // TODO Auto-generated method stub

    }

}

File: dota-zone/app/src/main/java/com/br/dotazone/view/adapter/FeedNewsListAdapter.kt
package com.br.dotazone.view.adapter

import android.content.Context
import android.content.Intent
import android.graphics.Typeface
import android.net.Uri
import android.text.Html
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView
import com.br.dotazone.DotaZoneBrain.TAG
import com.br.dotazone.DotaZoneBrain.rssItems
import com.br.dotazone.R
import com.br.dotazone.model.listeners.Controllable


class FeedNewsListAdapter(private val mContext: Context) : BaseAdapter(), Controllable {
	private val mInflate: LayoutInflater
	override fun getCount(): Int {
		Log.d(TAG, "getCount  " + rssItems!!.size + "")
		return rssItems!!.size
	}

	override fun getItem(position: Int): Any {
		Log.d(TAG, "getItem  $position")
		return rssItems!![position]
	}

	override fun getItemId(positionId: Int): Long {
		Log.d(TAG, "getItemId  $positionId")
		return positionId.toLong()
	}

	override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
		var view = convertView
		Log.d("teste", "position in view  $position")
		if (view == null) {
			view = mInflate.inflate(R.layout.feed_news_item, parent, false)
		}
		val font = Typeface.createFromAsset(mContext.assets, "Roboto-Light.ttf")
		val titleNews = view?.findViewById<View>(R.id.textItemTitle) as TextView
		titleNews.text = rssItems!![position].title
		val hourNews = view.findViewById<View>(R.id.textItemHour) as TextView
		hourNews.text = rssItems!![position].pubDate
		val descriptionNews = view.findViewById<View>(R.id.textItemDescription) as TextView
		descriptionNews.text = Html.fromHtml(rssItems!![position].description)
		descriptionNews.setOnClickListener {
			val browserIntent = Intent(Intent.ACTION_VIEW, Uri.parse(rssItems!![position].link))
			mContext.startActivity(browserIntent)
		}
		titleNews.setTypeface(font)
		hourNews.setTypeface(font)
		descriptionNews.setTypeface(font)
		return view
	}

	override fun menu(isOpen: Boolean) {
		// TODO Auto-generated method stub
	}

	init {
		mInflate = mContext.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
	}
}
